### PR TITLE
Need holistic analysis

### LINKED ISSUE #63: Need for holistic analysis
The discussion on traffic analysis is a little lightweight, but probably adequate for the scope of the document.  Basically, the problem is that certain information can leak (i.e., insufficient partitioning) in ways that can be subject to inference attacks.  It is hard to conceptualize this leakage in way that might allow you to conclude either that a system as a whole is acceptable or not.

It is possible to construct scenarios where the resolving power of traffic analysis might be considered too risky, even with separation.  For example, take an OHTTP deployment where the payload of the message is extremely interesting to an adversary.  Perhaps it contains a password, but not the identity of the account that it unlocks.  You might conclude that such a system is unacceptable for several reasons.  The space of accounts might be sufficiently small that simply trying the password with each in turn is justifiable to an attacker.  But even if the space of accounts was too large to test in that fashion, you might still reach that conclusion on the basis that the leakage from timing and other side channels is enough to allow an attacker to narrow the space of possible accounts down to a searchable size.

What this illustrates is that you probably need more tools than separation to make sense of the true privacy characteristics of a system.  For instance, you might look at the anonymity set that is created through the lens of something like k-anonymity; you might look information accumulation at the adversary in the strictly information theoretic sense; you might use differential privacy as the basis of an analysis; or, you might use symbolic analysis to examine the flow of data through a complex system.  Each approach might give you a different means of understanding the privacy characteristics of the system in a concrete fashion.  Without being able to connect your partitioning to specific outcomes, it is hard to make concrete statements about whether your privacy protections have been effective.

I don't think that this documents needs to be completely rigorous, but it might be worth saying something about how partitioning is a tool that can be used to manage privacy risks as part of a more comprehensive strategy.  But that you need further analysis to determine whether partitioning is the right tool and whether it is being effectively applied.  The examples given in the document include some of that analysis, but it's a little ad hoc.  That's OK, but it would be better if that text were more explicit about the need for this sort of holistic analysis.

### PR DESCRIPTION
**Pull request by chris-wood**
Closes #63

### CODE REVIEW

File: draft-iab-privacy-partitioning.md [hunk 1]
```diff
@@ -702,27 +710,31 @@ enough parties to recover identities.
 
 ## Violations by Insufficient Partitioning
 
-It is possible to define contexts that contain more than one type of user-specific information,
-despite efforts to do otherwise. As an example, consider OHTTP used for the purposes of hiding
+Partitioning alone is not a panacea for privacy. It is possible to apply partitioning in a way
```
Comment (tfpauly) [inline draft-iab-privacy-partitioning.md:713]: I wonder if we've used panacea too much at this point =)

File: draft-iab-privacy-partitioning.md [hunk 2]
```diff
@@ -664,8 +669,11 @@ solutions to the anti-abuse problem.
 
 # Limits of Privacy Partitioning {#limits}
 
-Privacy Partitioning aims to increase user privacy, though as stated, it is not a panacea.
-The privacy properties depend on numerous factors, including, though not limited to:
+Privacy partitioning aims to increase user privacy, though as stated, it is not a panacea.
+Privacy partitioning is an architectural tool that helps manage privacy risks. Understanding
+the limits of its benefits requires a more comprehensive analysis of the system in question.
+Such analysis also helps determine whether or not the tool has been applied correctly. In particular,
+the value of Privacy partitioning depends on numerous factors, including, though not limited to:
```
Comment (tfpauly) [inline draft-iab-privacy-partitioning.md:676]:
```suggestion
the value of privacy partitioning depends on numerous factors, including, though not limited to:
```

File: draft-iab-privacy-partitioning.md [hunk 3]
```diff
@@ -702,27 +710,31 @@ enough parties to recover identities.
 
 ## Violations by Insufficient Partitioning
 
-It is possible to define contexts that contain more than one type of user-specific information,
-despite efforts to do otherwise. As an example, consider OHTTP used for the purposes of hiding
+Partitioning alone is not a panacea for privacy. It is possible to apply partitioning in a way
+that is insufficient for meaningful privvacy. For example, partitioning at one layer in the stack
```
Comment (tfpauly) [inline draft-iab-privacy-partitioning.md:714]:
```suggestion
that is insufficient for meaningful privacy. For example, partitioning at one layer in the stack
```

File: draft-iab-privacy-partitioning.md (hunk 1 above)
Comment (chris-wood) [reply draft-iab-privacy-partitioning.md]: Heh, maybe =) I removed two of the three usages.

File: draft-iab-privacy-partitioning.md [hunk 4]
```diff
@@ -664,8 +669,11 @@ solutions to the anti-abuse problem.
 
 # Limits of Privacy Partitioning {#limits}
 
-Privacy Partitioning aims to increase user privacy, though as stated, it is not a panacea.
-The privacy properties depend on numerous factors, including, though not limited to:
+Privacy partitioning aims to increase user privacy, though as stated, it is merely one of possibly many
+architectural tool that help manage privacy risks. Understanding
```
Comment (tfpauly) [inline draft-iab-privacy-partitioning.md:673]:
```suggestion
architectural tools that help manage privacy risks. Understanding
```

**Review by tfpauly — APPROVED** [review]

File: draft-iab-privacy-partitioning.md [hunk 5]
```diff
@@ -700,29 +708,34 @@ be minimized and anonymized to prevent it from being useful for collusion.
...
-despite efforts to do otherwise. As an example, consider OHTTP used for the purposes of hiding
+Partitioning at the network layer can be insufficient when the application layer fails to properly
+partition. As an example, consider OHTTP used for the purposes of hiding
 client-identifying information for a browser telemetry system. It is entirely possible for reports
 in such a telemetry system to contain both client-specific telemetry data, such as information
 about their specific browser instance, as well as client-identifying information, such as the client's
-location or IP address. Even though OHTTP separates the client IP address from the server via
-a relay, the server still learns this directly from the client.
+email address, location, or IP address. Even though OHTTP separates the client IP address from the
+server via a relay, the server still can still this directly from the client's telemetry report.
```
Comment (mirjak) [inline draft-iab-privacy-partitioning.md:727]:
```suggestion
server via a relay, the server can still this directly from the client's telemetry report.
```

**Review by mirjak — APPROVED** [review]
This looks all good. Thanks! I wonder if we also need to say more explicitly that privacy partitioning is only one tool in the toolbox? Or is that clear anyway?

**Comment by tfpauly** [comment]
I think this makes it clear?

> Privacy partitioning aims to increase user privacy, though as stated, it is merely one of possibly many
>  architectural tools that help manage privacy risks. Understanding

**Pull request merged**